contributions/api/utils/datasetutils.py

```diff
@@ -0,0 +1,405 @@
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import copy
+
+from models.capabilities.contact import Contact
+from models.capabilities.datadeletionendpointdetail import DataDeletionEndpointDetail
+from models.capabilities.deploymentdetails import DeploymentDetails
+from models.capabilities.environmentvariable import EnvironmentVariable
+from models.talents.selfcertification import SelfCertification
+
+"""
+update contribution dataset
+"""
+def update_contribution_dataset_from_json(dataset, injson):
+    outjson = copy.copy(injson)
+    try:
+        dataset.set_name(injson['name'])
+        del outjson['name']
+    except:
+        pass
+    try:
+        dataset.set_short_description(injson['shortDescription'])
+        del outjson['shortDescription']
+    except:
+        pass
+    try:
+        dataset.set_long_description(injson["longDescription"])
+    except:
+        pass
+    try:
+        dataset.set_contributors(injson['contributors'])
+        del outjson['contributors']
+    except:
+        pass
+    try:
+        dataset.set_capabilities(injson['capabilities'])
+        del outjson['capabilities']
+    except:
+        pass
+    try:
+        dataset.set_talents(injson['talents'])
+        del outjson['talents']
+    except:
+        pass
+    try:
+        dataset.set_date_created(injson["dateCreated"])
+        del outjson["dateCreated"]
+    except Exception as e:
+        pass
+    try:
+        dataset.set_date_modified(injson["dateModified"])
+        del outjson["dateModified"]
+    except Exception as e:
+        pass
+
+    return dataset, outjson
+
+"""
+set capability dataset
+"""
+def update_capability_dataset_from_json(dataset, injson):
+    outjson = copy.copy(injson)
+    try:
+        dataset.set_name(injson['name'])
+        del outjson['name']
+    except:
+        pass
+    try:
+        dataset.set_description(injson['description'])
+        del outjson['description']
+    except:
+        pass
+    try:
+        dataset.set_is_open_source(injson['isOpenSource'])
+        del outjson['isOpenSource']
+    except:
+        pass
+    try:
+        dataset.set_api_doc_url(injson['apiDocUrl'])
+        del outjson['apiDocUrl']
+    except:
+        pass
+    try:
+        deployment_details = DeploymentDetails()
+        try:
+            location = injson["deploymentLocation"]["location"]
+            deployment_details.set_location(location)
+        except:
+            pass
+        try:
+            docker_image_name = injson["deploymentLocation"]["dockerImageName"]
+            deployment_details.set_docker_image_name(docker_image_name)
+        except:
+            pass
+        try:
+            environment_variable = EnvironmentVariable()
+            key = injson["deploymentLocation"]["environmentVariable"]["key"]
```

Comment:
Environment variables is a list of dictionary containing `key`, `value` pairs. This needs to be updated.